RecoEgamma/ElectronIdentification/plugins/ElectronIdMVABased.cc

```diff
@@ -114,41 +120,38 @@ ElectronIdMVABased::~ElectronIdMVABased()
 
 // ------------ method called on each new Event  ------------
 bool ElectronIdMVABased::filter(edm::Event& iEvent, const edm::EventSetup& iSetup) {
-	using namespace edm;
-
-	std::auto_ptr<reco::GsfElectronCollection> mvaElectrons(new reco::GsfElectronCollection);
-
-	Handle<reco::VertexCollection>  vertexCollection;
-	iEvent.getByToken(vertexToken, vertexCollection);
-	int nVtx = vertexCollection->size();
-
-	Handle<reco::GsfElectronCollection> egCollection;
-	iEvent.getByToken(electronToken,egCollection);
-	const reco::GsfElectronCollection egCandidates = (*egCollection.product());
-	for ( reco::GsfElectronCollection::const_iterator egIter = egCandidates.begin(); egIter != egCandidates.end(); ++egIter) {
-	  double mvaVal = mvaID_->mva( *egIter, nVtx );
-	  double isoDr03 = egIter->dr03TkSumPt() + egIter->dr03EcalRecHitSumEt() + egIter->dr03HcalTowerSumEt();
-	  double eleEta = fabs(egIter->eta());
-	  if (eleEta <= 1.485 && mvaVal > thresholdBarrel && isoDr03 < thresholdIsoBarrel) {
-	    mvaElectrons->push_back( *egIter );
-	    reco::GsfElectron::MvaOutput myMvaOutput;
-	    myMvaOutput.mva_Isolated = mvaVal;
-	    mvaElectrons->back().setMvaOutput(myMvaOutput);
-	  }
-	  else if (eleEta > 1.485 && mvaVal > thresholdEndcap  && isoDr03 < thresholdIsoEndcap) {
-	    mvaElectrons->push_back( *egIter );
-	    reco::GsfElectron::MvaOutput myMvaOutput;
-	    myMvaOutput.mva_Isolated = mvaVal;
-	    mvaElectrons->back().setMvaOutput(myMvaOutput);
-	  }
-
-
-	}
-
-
-	iEvent.put(mvaElectrons);
-
-	return true;
+  using namespace edm;
+  
+  std::auto_ptr<reco::GsfElectronCollection> mvaElectrons(new reco::GsfElectronCollection);
+  
+  Handle<reco::VertexCollection>  vertexCollection;
+  iEvent.getByToken(vertexToken, vertexCollection);
+  int nVtx = vertexCollection->size();
+  
+  Handle<reco::GsfElectronCollection> egCollection;
+  iEvent.getByToken(electronToken,egCollection);
+  const reco::GsfElectronCollection egCandidates = (*egCollection.product());
+  for ( reco::GsfElectronCollection::const_iterator egIter = egCandidates.begin(); egIter != egCandidates.end(); ++egIter) {
+    double mvaVal = globalCache()->mvaID_->mva( *egIter, nVtx );
+    double isoDr03 = egIter->dr03TkSumPt() + egIter->dr03EcalRecHitSumEt() + egIter->dr03HcalTowerSumEt();
+    double eleEta = fabs(egIter->eta());
+    if (eleEta <= 1.485 && mvaVal > thresholdBarrel && isoDr03 < thresholdIsoBarrel) {
+      mvaElectrons->push_back( *egIter );
+      reco::GsfElectron::MvaOutput myMvaOutput;
+      myMvaOutput.mva_Isolated = mvaVal;
+      mvaElectrons->back().setMvaOutput(myMvaOutput);
+    }
+    else if (eleEta > 1.485 && mvaVal > thresholdEndcap  && isoDr03 < thresholdIsoEndcap) {
```

Comment:
It might be good to replace the magic number 1.485 with a constant (on line 138, too) when there is an opportunity.